distributed/utils.py

```diff
@@ -1426,6 +1426,45 @@ def clean_dashboard_address(addrs: AnyType, default_listen_ip: str = "") -> List
     return addresses
 
 
+@functools.lru_cache(maxsize=256)
+def _clean_filename(filename):
+    """Removes potentially sensitive information from a Python module file path"""
+
+    if filename.startswith(sys.exec_prefix):
+        filename = filename.replace(sys.exec_prefix, "...")
+        if "site-packages" not in filename:
+            # Python standard library module
+            return filename
+        else:
+            # TODO: Use `sys.platlibdir` once Python 3.9 is the minimum supported python version
+            if filename.startswith(os.path.join("...", "lib64")):
+                platlibdir = "lib64"
+            else:
+                platlibdir = "lib"
+            if os.sep == "\\":
+                return filename.replace(
+                    os.path.join(
+                        os.sep + platlibdir,
+                        "site-packages",
+                    ),
+                    "",
+                )
+            else:
+                return filename.replace(
+                    os.path.join(
+                        os.sep + platlibdir,
+                        f"python{sys.version_info[0]}.{sys.version_info[1]}",
+                        "site-packages",
+                    ),
+                    "",
+                )
```

Comment:
There was a comment about us not dealing with version numbers and path reconstruction, etc. I don't mind going ahead with this since it's been sitting for so long. However, a rather simple and possibly generic implementation would be

```python

In [1]: import os

In [2]: def func(filename):
   ...:     filename_cut = filename.split("site-packages")[-1]
   ...:     filename_stripped = filename_cut.replace(os.sep, "", 1 if filename.startswith(os.sep) else 0)
   ...:     return filename_stripped
   ...:

In [3]: func("foo")
Out[3]: 'foo'

In [4]: func("/Users/fjetter/mambaforge/envs/dask-distributed/lib/python3.8/site-packages/pandas/core/computation/expr.py")
Out[4]: 'pandas/core/computation/expr.py'

In [5]: func("~/my_very_own_directory/foo/bar")
Out[5]: '~/my_very_own_directory/foo/bar'

In [6]: func("/root/is/stripped/if/no/site-package/")
Out[6]: 'root/is/stripped/if/no/site-package/'
```

the only assumption is that the paths are sitting in site-packages but that's it.

Reply:
I think the problem with that is that for windows the paths are not in  `site-packages`, or at least I remember a conversation about this. I can take a look at it and see if that's the case.